packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -744,6 +749,15 @@ class RenderParagraph extends RenderBox
     return _textPainter.getOffsetForCaret(position, caretPrototype);
   }
 
+  /// Returns the tight bounded height of the glyph at the given [position].
+  ///
+  /// Valid only after [layout].
+  double? getFullHeightForCaret(TextPosition position, Rect caretPrototype) {
```

Comment:
This should be ok to expose, as the CaretMetrics construct is native to TextPainter.

The name "fullHeight" seems a bit ambiguous in hindsight especially since there is no API for not-full height here. We may want to consider renaming it to something like "rawHeight" here, indicating it is unprocessed and a direct result of the caret computation. The alternative is to just keep the naming as "full" for consistency.

I am leaning towards keeping it as "full" as diverging API is not good. @justinmc any opinions?

Reply:
Is there anything else that refers to "full height" in the codebase?  Not that I could think of or find with a quick search.

If it's really just this method, maybe we should nip it in the bud and switch to "rawHeight" while we have the chance?  If I'm wrong then it definitely makes sense to be consistent and use "fullHeight".